src/LoggingFormats.jl

```diff
@@ -1,5 +1,40 @@
 module LoggingFormats
 
-greet() = print("Hello World!")
+export Truncated
+
+using Logging
+
+function shorten_str(str, max_len)
+    suffix = "…"
+    if length(str) > max_len
+        str[1:nextind(str, 0, min(end, max_len-length(suffix)))] * suffix
+    else
+        str
+    end
+end
+
+struct Truncated <: Function
+    max_var_len::Int
+    Truncated(max_var_len) = max_var_len <= 0 ? error("max_var_len must be positive") : new(max_var_len)
+end
+Truncated() = Truncated(5_000)
+
+# copied from https://github.com/JuliaLang/julia/blob/v1.5.4/stdlib/Logging/src/ConsoleLogger.jl and modified
+function (tr::Truncated)(io, args)
+    levelstr = args.level == Logging.Warn ? "Warning" : string(args.level)
+    msglines = split(chomp(string(shorten_str(args.message, tr.max_var_len))), '\n')
```

Comment:
`shorten_str` should already return a string:
```suggestion
    msglines = split(chomp(shorten_str(args.message, tr.max_var_len)), '\n')
```